ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/stack/StackComponentView.kt

```diff
@@ -171,6 +189,112 @@ private fun StackWithOverlaidBadge(
     }
 }
 
+@Suppress("LongMethod", "LongParameterList", "CyclomaticComplexMethod")
+@Composable
+private fun StackWithEdgeToEdgeBadge(
+    stackState: StackComponentState,
+    state: PaywallState.Loaded.Components,
+    badgeStack: StackComponentStyle,
+    alignment: TwoDimensionalAlignment,
+    clickHandler: suspend (PaywallAction) -> Unit,
+    modifier: Modifier = Modifier,
+    selected: Boolean = false,
+) {
+    var badgeHeight by remember {
+        mutableIntStateOf(0)
+    }
+    val badgeHeightDp = with(LocalDensity.current) { badgeHeight.toDp() }
+    val badge = @Composable {
+        StackComponentView(
+            // The background and border is applied to the parent container so we null it out here.
+            // We make the badge use all the available width without increasing the size of the main content.
+            badgeStack.copy(
+                backgroundColor = null,
+                size = Size(width = Fill, height = badgeStack.size.height),
+                border = null,
+                margin = PaddingValues(0.dp),
+            ),
+            state,
+            clickHandler,
+            selected = selected,
+            modifier = Modifier.onGloballyPositioned {
+                badgeHeight = it.size.height
+            },
+        )
+    }
+    val mainContent = @Composable {
+        MainStackComponent(
+            stackState,
+            state,
+            clickHandler,
+            selected = selected,
+        )
+    }
+    val backgroundColorStyle = badgeStack.backgroundColor?.let { rememberColorStyle(scheme = it) }
+    val borderStyle = badgeStack.border?.let { rememberBorderStyle(border = it) }
+    val shadowStyle = badgeStack.shadow?.let { rememberShadowStyle(shadow = it) }
+    val backgroundShape = badgeStack.cornerRadiuses?.let { cornerRadiuses ->
+        val filteredCornerRadiuses = when (alignment) {
+            TwoDimensionalAlignment.TOP_LEADING,
+            TwoDimensionalAlignment.TOP,
+            TwoDimensionalAlignment.TOP_TRAILING,
+            -> cornerRadiuses.copy(bottomLeading = 0.0, bottomTrailing = 0.0)
+            TwoDimensionalAlignment.BOTTOM_LEADING,
+            TwoDimensionalAlignment.BOTTOM,
+            TwoDimensionalAlignment.BOTTOM_TRAILING,
+            -> cornerRadiuses.copy(topLeading = 0.0, topTrailing = 0.0)
```

Comment:
Here I'm clearing some of the corner radiuses for the badge, so it reaches the expected point.